src/main/java/com/mongodb/hibernate/dialect/MongoDialect.java

```diff
@@ -16,17 +16,21 @@
 
 package com.mongodb.hibernate.dialect;
 
+import com.mongodb.hibernate.jdbc.MongoConnectionProvider;
 import org.hibernate.dialect.DatabaseVersion;
 import org.hibernate.dialect.Dialect;
 import org.hibernate.engine.jdbc.dialect.spi.DialectResolutionInfo;
 
 /**
- * A MongoDB {@link Dialect} for {@linkplain #getMinimumSupportedVersion() version 6.0 and above}.
+ * A MongoDB {@link Dialect} for {@linkplain #getMinimumSupportedVersion() version 6.0 and above}. Must be used together
+ * with {@link MongoConnectionProvider}.
```

Comment:
not sure why we need to emphasize the combination. To me the two classes are not tightly coupled. Traditionally `Dialect` is used to customize programmatically (as opposed to `hibernate.properties` config file). `ConnectionProvider` customization is in the scope of properties file without corresponding Dialect method to be overridden.

But it is ok to retain it. Now I think your point is the two mandatory entries in hibernate.properties:
```
hibernate.dialect=com.mongodb.hibernate.dialect.MongoDialect
hibernate.connection.provider_class=com.mongodb.hibernate.jdbc.MongoConnectionProvider
```

Reply:
Discussed. This is fine.